lib/manageiq_performance/middleware_storage/memory.rb

```diff
@@ -0,0 +1,35 @@
+module ManageIQPerformance
+  module MiddlewareStorage
+    class Memory
+      attr_reader :current_run, :profile_captures
+
+      def initialize
+        @current_run = {}
+        @profile_captures = []
+        ManageIQPerformance.last_run = nil
+      end
+
+      def record _, tag, __, data
+        @current_run[tag] = data.call
+      end
+
+      def finalize
+        @profile_captures << @current_run
+        ManageIQPerformance.last_run = @current_run
+        @current_run = {}
+      end
+    end
+  end
+
+  @memory_storage_mutex = Mutex.new
+  @memory_storage_mutex.synchronize { @last_run = nil }
```

Comment:
> The mutex is a class level variable?

Yes, this is writing a method to the top level `ManageIQPerformance` module, so you can go `ManageIQPerformance.last_run`.

> Can you just call `last_run=` instead?

Actually tried that, but got a different error (first error):

```console
$ rspec --seed 52889
/Users/nicklamuro/code/redhat/manageiq-performance/lib/manageiq_performance/middleware_storage/memory.rb:32: warning: assigned but unused variable - last_run

Randomized with seed 52889
............................................................................................................................................................./Users/nicklamuro/code/redhat/manageiq-performance/lib/manageiq_performance/middleware_storage/memory.rb:29: warning: instance variable @last_run not initialized
.......

Finished in 1.74 seconds (files took 1.05 seconds to load)
164 examples, 0 failures

Randomized with seed 52889
```